### PR TITLE
allowing globs for situations where we load files dynamically from a folder

### PR DESCRIPTION
**Pull request by babymechanic**
@nguyenchr @rprieto @TabDigital/ping-api

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -65,21 +66,29 @@ exports.lint = function(opts) {
...
-  }
-  if (pkg.bin) {
-    sources = sources.concat(_.values(pkg.bin));
-  }
-  if (additional) {
-    sources = sources.concat(additional);
-  }
-  return sources.map(function(file){
+  return listify(pkg.main)
+  .concat(_.values(pkg.bin? pkg.bin : [] ))
```
Comment (rprieto) [inline lib/index.js]: Could this be `.concat(_.value(pkg.bin))` ? It seems it returns `null`, `undefined` and `{}` all return an empty array.

File: lib/index.js [hunk 2]
```diff
@@ -65,21 +66,29 @@ exports.lint = function(opts) {
...
 }
 
+var getAdditionalSources = function(basedir, additionalSources){
+  var files = [];
+  listify(additionalSources)
+  .forEach(function(additionalSource){
+    files = files.concat(glob.sync(additionalSource,{cwd: basedir,nodir:true}));
+  });
+  return files;
+};
```
Comment (rprieto) [inline lib/index.js]: What about something like (untested :smile:)

```js
return listify(additionalSources).reduce(function(acc, pattern) {
  var matches = glob.sync(pattern,{cwd: basedir, nodir:true});
  return acc.concat(pattern);
})
```

**Comment by rprieto** [comment]
Thanks, a few comments but otherwise looks great!
Will be very useful :+1:

File: lib/index.js (hunk 1 above)
Comment (babymechanic) [inline lib/index.js]: my bad, i wanted to return an empty object. thanks for catching that

Comment (rprieto) [inline lib/index.js]: sorry I meant `_.value(undefined)` and `_.value({})` should both behave OK, if `bin` is empty.

**Pull request merged**